src/thermal_conductivity/scattering.f90

```diff
@@ -0,0 +1,347 @@
+#include "precompilerdefinitions"
+module scattering
+use konstanter, only: r8, i8, lo_freqtol, lo_twopi, lo_exitcode_param, lo_hugeint, lo_pi, lo_tol, &
+                      lo_phonongroupveltol, lo_tol, lo_frequency_THz_to_Hartree, lo_kb_hartree, lo_huge
+use gottochblandat, only: walltime, lo_trueNtimes, lo_progressbar_init, lo_progressbar, lo_gauss, lo_planck, lo_return_unique
+use mpi_wrappers, only: lo_mpi_helper, lo_stop_gracefully
+use lo_memtracker, only: lo_mem_helper
+use type_crystalstructure, only: lo_crystalstructure
+use type_forceconstant_secondorder, only: lo_forceconstant_secondorder
+use type_forceconstant_thirdorder, only: lo_forceconstant_thirdorder
+use type_forceconstant_fourthorder, only: lo_forceconstant_fourthorder
+use type_qpointmesh, only: lo_qpoint_mesh, lo_fft_mesh
+use type_phonon_dispersions, only: lo_phonon_dispersions
+use type_symmetryoperation, only: lo_operate_on_vector
+use lo_randomnumbers, only: lo_mersennetwister
+use lo_fftgrid_helper, only: lo_montecarlo_grid, singlet_to_triplet, triplet_to_singlet, &
+                             fft_third_grid_index, fft_fourth_grid_index
+use lo_timetracker, only: lo_timer
+
+use options, only: lo_opts
+
+use lo_sorting, only: lo_qsort
+
+implicit none
+private
+public :: lo_scattering_rates
+
+real(r8), parameter :: isotope_prefactor = lo_pi/4.0_r8
+real(r8), parameter :: threephonon_prefactor = lo_pi/16.0_r8
+real(r8), parameter :: fourphonon_prefactor = lo_pi/96.0_r8
+
+! Container for scattering rates
+type lo_scattering_rates
+    !> The number of qpoint/mode on this rank
+    integer :: nlocal_point
+    !> The list of qpoint and modes for this rank
+    integer, dimension(:), allocatable :: q1, b1
+    !> Let's precompute the Bose-Einstein distribution
+    real(r8), dimension(:, :), allocatable :: be, sigsq
```

Comment:
I can make the name more explicit

For the `sigsq` we don't necessarily need to make it squared here, but I'm doing it this way because we only need it squared. So it was to avoid squaring it every time